File: Python/Program/SivitasAkademik.py
# Kelas Sivitas Akademik
class SivitasAkademik():
    # Atribut Mahasiswa:
    __asal_universitas = ""
    __email_edu = ""
    # Atribut Komponen:
    __mahasiswa = []
    __dosen = []

    # Konstruktor
    def __init__(self, asal_universitas, email_edu, mahasiswa, dosen):
        self.__asal_universitas = asal_universitas
        self.__email_edu = email_edu
        self.__mahasiswa = mahasiswa.copy()
        self.__dosen = dosen.copy()

    # Setter
    def set_asal_universitas(self, asal_universitas):
        self.__asal_universitas = asal_universitas

    def set_email_edu(self, email_edu):
        self.__email_edu = email_edu

    def set_mahasiswa(self, mahasiswa):
        self.__mahasiswa = mahasiswa.copy()

    def set_dosen(self, dosen):
        self.__dosen = dosen.copy()

    # Getter
    def get_asal_universitas(self):
        return self.__asal_universitas
    
    def get_email_edu(self):
        return self.__email_edu
    
    def get_mahasiswa(self):
        return self.__mahasiswa
    
    def get_dosen(self):
        return self.__dosen

File: CPP/Program/SivitasAkademik.cpp
// Import file:
#include "header.hh"

// Empty konstruktor.
SivitasAkademik::SivitasAkademik()
{
    this->asal_universitas = "";
    this->email_edu = "";
    this->mahasiswa = {};
    this->dosen = {};
}

// Konstruktor with params.
SivitasAkademik::SivitasAkademik(string asal_universitas, string email_edu, vector<Mahasiswa> mahasiswa, vector<Dosen> dosen)
{
    this->asal_universitas = asal_universitas;
    this->email_edu = email_edu;
    this->mahasiswa = mahasiswa;
    this->dosen = dosen;
}

// Setter
void SivitasAkademik::set_asal_universitas(string asal_universitas)
{
    this->asal_universitas = asal_universitas;
}
void SivitasAkademik::set_email_edu(string email_edu)
{
    this->email_edu = email_edu;
}
void SivitasAkademik::set_mahasiswa(vector<Mahasiswa> mahasiswa)
{
    this->mahasiswa = mahasiswa;
}
void SivitasAkademik::set_dosen(vector<Dosen> dosen)
{
    this->dosen = dosen;
}

// Getter
string SivitasAkademik::get_asal_universitas()
{
    return this->asal_universitas;
}
string SivitasAkademik::get_email_edu()
{
    return this->email_edu;
}
vector<Mahasiswa> SivitasAkademik::get_mahasiswa()
{
    return this->mahasiswa;
}
vector<Dosen> SivitasAkademik::get_dosen()
{
    return this->dosen;
}

SivitasAkademik::~SivitasAkademik()
{
    
}